jsk_arc2017_baxter/euslisp/lib/apc-interface.l

```diff
@@ -0,0 +1,376 @@
+      (send self :wait-interpolation)
+      (unix::sleep 1)  ;; wait for arm to follow
+      (setq graspingp (send self :graspingp arm))
+      (ros::ros-info "[:try-to-pick-object-in-bin] arm:~a graspingp: ~a" arm graspingp)
+    graspingp))
+  (:ik->bin-center
+    (arm bin &key (offset #f(0 0 0)) (coords-pitch 0) (rotation-axis t) (use-gripper nil))
+    (let (bin-box bin-coords)
+      (setq bin-box (gethash bin bin-boxes-))
+      (setq bin-coords (send tfl- :tf-pose->coords
+                             (send (send bin-box :header) :frame_id)
+                             (send bin-box :pose)))
+      (send bin-coords :translate
+            (v+ (float-vector
+                  0.0 0.0 (/ (m->mm (send bin-box :dimensions :z)) 2.0))
+                 offset) :world)
+      (send bin-coords :rotate (* pi (/ coords-pitch 180)) :y)
+      (send *baxter* arm :inverse-kinematics bin-coords
+            :rotation-axis rotation-axis
+            :use-gripper use-gripper)))
+  (:ik->bin-entrance
+    (arm bin &key (offset #f(0 0 0)) (coords-pitch 0) (rotation-axis t) (use-gripper nil))
+    (let (bin-box bin-coords)
+      (setq bin-box (gethash bin bin-boxes-))
+      (setq bin-coords (send tfl- :tf-pose->coords
+                             (send (send bin-box :header) :frame_id)
+                             (send bin-box :pose)))
+      (send bin-coords :translate
+            (v+ (float-vector
+                  (- (/ (m->mm (send bin-box :dimensions :x)) 2.0))
+                  0.0 (/ (m->mm (send bin-box :dimensions :z)) 2.0))
+                offset) :world)
+      (send bin-coords :rotate (* pi (/ coords-pitch 180)) :y)
+      (send *baxter* arm :inverse-kinematics bin-coords
+            :rotation-axis rotation-axis
+            :use-gripper use-gripper)))
+  (:ik->cardboard-entrance
+    (arm cardboard &key (offset #f(0 0 0)) (rotation-axis t) (use-gripper nil))
+    (send *baxter* arm :inverse-kinematics
+          (make-coords
+            :pos (v+ offset
+                     (cond ((eq cardboard :a) (float-vector 950 500 0))
+                           ((eq cardboard :b) (float-vector 650 500 0))
+                           (t (float-vector 800 -500 0))))
+            :rpy (float-vector 0 0 0))
+          :rotation-axis rotation-axis
+          :use-gripper use-gripper))
+  (:move-arm-body->bin-overlook-pose
+    (arm bin &key (gripper-angle 90))
+    (let (avs)
+      (setq avs (list (send *baxter* :avoid-shelf-pose arm (if (eq arm :larm) :d :f))))
+      (send *baxter* :reset-pose arm)
+      (send *baxter* :rotate-gripper arm gripper-angle :relative nil)
+      (pushback
+        (send self :ik->bin-entrance arm bin
+              :offset #f(-100 0 300) :coords-pitch gripper-angle
+              :rotation-axis :y
+              :use-gripper nil) avs)
+      (send self :angle-vector-sequence avs :fast (get-arm-controller arm) 0 :scale 3.0))))
+
+(defun jsk_arc2017_baxter::apc-init (&key (ctype :default-controller) (moveit nil))
```

Comment:
Why not these initializations will be done in `jsk_arc_2017_baxter::apc-interface :init`?